Source/BugsnagDeviceWithState.m

```diff
@@ -7,7 +7,85 @@
 //
 
 #import "BugsnagDeviceWithState.h"
+#import "BugsnagCollections.h"
+#import "BugsnagLogger.h"
+
+NSNumber *BSGDeviceFreeSpace(NSSearchPathDirectory directory) {
+    NSNumber *freeBytes = nil;
+    NSFileManager *fileManager = [NSFileManager defaultManager];
+    NSArray *searchPaths = NSSearchPathForDirectoriesInDomains(directory, NSUserDomainMask, true);
+    NSString *path = [searchPaths lastObject];
+
+    NSError *error;
+    NSDictionary *fileSystemAttrs =
+            [fileManager attributesOfFileSystemForPath:path error:&error];
+
+    if (error) {
+        bsg_log_warn(@"Failed to read free disk space: %@", error);
+    } else {
+        freeBytes = [fileSystemAttrs objectForKey:NSFileSystemFreeSize];
+    }
+    return freeBytes;
+}
```

Comment:
Moving this function is a good opportunity to provide it with some documentation.

Reply:
Added a few docs